library/Graphite/Util/MacroTemplate.php

```diff
@@ -200,6 +205,22 @@ protected function getReverseResolvePattern()
             $result[] = '\z/s';
 
             $this->reverseResolvePattern = implode($result);
+
+            $searchReplace = [
+                '\[' => '[', //Don't quote [, -, ]
+                '\-' => '-',
+                '\]' => ']',
+                '\{' => '(', // change graphite list syntax to regex
+                ','  => '|',
+                '\}' => ')',
+                '\*' => '(.*)' // change graphite * syntax to regex
```

Comment:
Is this why you need to break out of the foreach above? Is this still required if you make this a non-capturing group?